### ISSUE TITLE
.T method mutates the object

### ISSUE THREAD

**Issue by vspinu** [DynamicTimeWarping/dtw-python#76]
**Describe the bug**

Currently `.T()` method mutates step pattern objects which might result in unexpected behavior, especially with built-in patterns. 

**To Reproduce**
```py
from dtw.stepPattern import asymmetric
print(asymmetric)
Step pattern recursion:
 g[i,j] = min(
     g[i-1,j  ] +     d[i  ,j  ] ,
     g[i-1,j-1] +     d[i  ,j  ] ,
     g[i-1,j-2] +     d[i  ,j  ] ,
 ) 
Normalization hint: N
tt = asymmetric.T()
print(asymmetric)
Step pattern recursion:
 g[i,j] = min(
     g[i  ,j-1] +     d[i  ,j  ] ,
     g[i-1,j-1] +     d[i  ,j  ] ,
     g[i-2,j-1] +     d[i  ,j  ] ,
 ) 
Normalization hint: M
```

**Version**

dtw: '1.3.0'

**Comment by vspinu** [comment]
If this is an expected behavior, maybe a copy method could be added instead?

**Comment by tonigi** [comment]
Ouch! Thanks for spotting it. No, step patterns should be immutable. Will
fix asap.

Il dom 22 ott 2023, 11:46 Vitalie Spinu ***@***.***> ha
scritto:

**Comment by tonigi** [comment]
Hopefully fixed in ae3b3657a1dd2d793fb5346e188826de15e0b4c1 and 1.3.1
